pretyping/evarsolve.ml

```diff
@@ -101,9 +102,9 @@ let refresh_universes ?(status=univ_rigid) ?(onlyalg=false) ?(refreshset=false)
   in
   let t' = 
     if isArity t then
-      (match pbty with
-      | None -> t
-      | Some dir -> refresh status dir t)
+      match pbty with
+      | None -> refresh true univ_flexible false t
```

Comment:
I'd like to understand this change, but given how hard it is to grasp everything that is around, I'd need some comments.